### PR TITLE
Team: Rework to work with class based Teams instead of strings

### PR DESCRIPTION
**Pull request by Alf21**

### CODE REVIEW

File: code/global/RPCs.cs [hunk 1]
```diff
@@ -52,7 +52,14 @@ public static void ClientSetRole(TTTPlayer player, string roleName, string teamN
                 return;
             }
 
-            player.SetRole(Utils.GetObjectByType<TTTRole>(Utils.GetTypeByName<TTTRole>(roleName)), TTTTeam.GetTeam(teamName));
+            TTTTeam team = null;
+
+            if (teamName != null)
+            {
+                team = TeamFunctions.GetTeam(teamName);
```
Comment (mzegar) [inline code/global/RPCs.cs:59]: Can we just handle the `teamName` null check inside of `GetTeam()`?

File: code/teams/TraitorTeam.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+namespace TTTReborn.Teams
+{
+    [TeamAttribute("Traitors")]
+    public class TraitorTeam : TTTTeam
+    {
+        public override Color Color => Color.FromBytes(223, 41, 53);
```
Comment (mzegar) [inline code/teams/TraitorTeam.cs:6]: Can we just use regular R, G, B values here?

**Review by mzegar — COMMENTED** [review]
Quick comments

File: code/teams/TraitorTeam.cs (hunk 2 above)
Comment (Alf21) [reply code/teams/TraitorTeam.cs:6]: it is RGB. What do you mean? :)

**Review by mzegar — APPROVED** [review]
LGTM

**Pull request merged**